File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        vector<int> v(26, 0);

        int i = 0;        
        int ans = 0;

        for (int j=0; j<s.size(); j++){
            int ind =  s[j] - 'a';
            v[ind]++;

            while (v[ind] > 2) {
                v[s[i] - 'a'] -- ;
                i++;
            }

            ans = max (ans, j-i+1);
        }

        return ans;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int[] arr = new int[26];

        int i=0;
        int ans=0;

        for (int j=0; j<s.length(); j++){
            int ind = s.charAt(j) - 'a';
            arr[ind]++;

            while (arr[ind] > 2) {
                arr[s.charAt(i) - 'a']--;
                i++;
            }

            ans = Math.max(ans, j-i+1);
        }

        return ans;
    }
}